### ISSUE TITLE
Document ignoring urls within comments in max-line-length

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#2630]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A feature request to ignore URLs in `max-line-length` rule comments

> Which rule, if any, is this issue related to?

 `max-line-length`

> What CSS is needed to reproduce this issue?

```css
/* ignore urls https://www.example.com */
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "max-line-length": [ 20, {
      "ignore": "non-comments",
    } ]
  }
}
```

> Which version of stylelint are you using?

`7.11.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config .stylelintrc *.css`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, would need to include support for other non-standard syntaxes

> What did you expect to happen?

URLs to not be counted in the line length

> What actually happened (e.g. what warnings or errors you are getting)?

The character length of the URL *is* included in the length count

```shell
test.css
 2:8  ✖  Expected line length to be no more than 20 characters   max-line-length
```

**Comment by jeddy3** [comment]
We currently collapse all types of urls (e.g. schemeless, path relative etc...) in non-comments by hooking into `url()` function.

I guess in comments we can [just look for](https://github.com/sindresorhus/linkify-urls/blob/master/index.js#L5) just `http(s)` urls and collapse them. This could be done with an option, say `collapseUrlsInComments: true`.

The decision is whether we add this option (as it's not nearly as clean as hooking into `url()`, or require the user to use disable comments to turn off the rule when necessary.

**Comment by evilebottnawi** [comment]
@jeddy3 why do not add `ignoreUrl` option?

**Comment by jeddy3** [comment]
> @jeddy3 why do not add ignoreUrl option?

`ignore*` options tend to target contructs of the language e.g. comments, at-rules, values, functions etc. In this instance, the request to discount urls within comment strings. For me, `ignoreUrl` implies ignoring the `url()` construct.

**Comment by evilebottnawi** [comment]
@jeddy3 hm, your are right, maybe `ignorePattern` is best solution for this? We can add example in docs how ignore urls  using `ignorePattern`.

**Comment by jeddy3** [comment]
> We can add example in docs how ignore urls using `ignorePattern`.

I was just thinking about that too :) SGTM. 

@ntwb Do you want to draw up a regex that spots urls within comment blocks? We can then add it as a second example to the [`ignorePattern: ["/regex/"]`](https://github.com/stylelint/stylelint/tree/master/lib/rules/max-line-length#ignorepattern-regex) option.

**Comment by ntwb** [comment]
Sounds good to me, thanks for spearheading the conversion

Let's go with a relative simple regex:  `ignorePattern: ["/https?:\/\/[0-9,a-z]*.*/"]`, there far more complex regex's out there for URLs but let's keep it simple.

I've added it to the docs in a new example in #2635

**Comment by ntwb** [comment]
I've just added a test for the example in #2636

**Comment by jeddy3** [comment]
Done in #2636 & #2635
